target/scripts/startup/setup.d/saslauthd.sh

```diff
@@ -1,29 +1,29 @@
 #!/bin/bash
 
-
 function _setup_saslauthd() {
   _log 'debug' 'Setting up SASLAUTHD'
 
-  if [[ ! -f /etc/saslauthd.conf ]]; then
+  # NOTE: It's unlikely this file would already exist,
+  # Unlike Dovecot/Postfix LDAP support, this file has no ENV replacement
+  # nor does it copy from the DMS config volume to this internal location.
+  if [[ ${ACCOUNT_PROVISIONER} == 'LDAP' ]] \
+  && [[ ! -f /etc/saslauthd.conf ]]; then
     _log 'trace' 'Creating /etc/saslauthd.conf'
-    cat > /etc/saslauthd.conf << EOF
-ldap_servers: ${SASLAUTHD_LDAP_SERVER}
-
-ldap_auth_method: ${SASLAUTHD_LDAP_AUTH_METHOD}
-ldap_bind_dn: ${SASLAUTHD_LDAP_BIND_DN}
-ldap_bind_pw: ${SASLAUTHD_LDAP_PASSWORD}
-
-ldap_search_base: ${SASLAUTHD_LDAP_SEARCH_BASE}
-ldap_filter: ${SASLAUTHD_LDAP_FILTER}
-
-ldap_start_tls: ${SASLAUTHD_LDAP_START_TLS}
-ldap_tls_check_peer: ${SASLAUTHD_LDAP_TLS_CHECK_PEER}
-
-${SASLAUTHD_LDAP_TLS_CACERT_FILE}
-${SASLAUTHD_LDAP_TLS_CACERT_DIR}
-${SASLAUTHD_LDAP_PASSWORD_ATTR}
-${SASLAUTHD_LDAP_MECH}
 
+    # Create a config based on ENV
+    sed '/^.*: $/d'> /etc/saslauthd.conf << EOF
```

Comment:
What does this `sed` actually do?

Reply:
`key: ` is matched (_space could probably be `\s*`_), so that if no value was assigned via the HereDoc input, the line is removed from the output.

Before we had an awkward workaround to avoid keys with empty values, which resulted in multiple blank lines if unset.

Reply:
Only adding lines without empty vars.

Reply:
I just noticed, for empty vars, ${foo:-} should have been used. ATM that's not a problem. But a good practice, if we want to introduce "set -u" in the future.